### PR TITLE
fix(ui-react): make recovery email validation case-insensitive

### PR DESCRIPTION
**Pull request by luizhf42**
## Summary

- Fix case-sensitive equality check in `validateRecoveryEmail` that allowed setting a recovery email identical to the primary email when using different casing (e.g., `User@Example.com` vs `user@example.com`)
- Extract `validateRecoveryEmail` into `pages/profile/validate.ts`, following the same pattern as the setup page
- Add unit tests covering empty input, invalid format, exact match, case-insensitive match, and valid distinct email

## Motivation

The UI was performing a strict equality check (`===`), which meant mixed-case duplicates bypassed client-side validation and only got caught by the API. This aligns the UI with the backend's case-insensitive behavior.

## Testing

1. Go to profile settings and set a recovery email matching the primary email with different casing — validation should reject it
2. Run `npx vitest run apps/admin/src/pages/profile/__tests__/validate.test.ts` — all 6 tests should pass

### CODE REVIEW

File: ui-react/apps/admin/src/pages/Profile.tsx [hunk 1]
```diff
@@ -41,10 +41,10 @@ function validateEmail(v: string): string | null {
...
 }
 
-function validateRecoveryEmail(v: string, primary: string): string | null {
-  if (!v) return null;
-  if (!/^[^\s@]+@[^\s@]+\.[^\s@]+$/.test(v)) return "Invalid email format";
-  if (v === primary) return "Must be different from your email";
+function validateRecoveryEmail(recoveryEmail: string, primaryEmail: string): string | null {
+  if (!recoveryEmail) return null;
+  if (!/^[^\s@]+@[^\s@]+\.[^\s@]+$/.test(recoveryEmail)) return "Invalid email format";
+  if (recoveryEmail.toLowerCase() === primaryEmail.toLowerCase()) return "Must be different from your email";
```
[the remark this answers is not in the document]
Comment (luizhf42) [reply ui-react/apps/admin/src/pages/Profile.tsx]: Moved `validateRecoveryEmail` to a separate `validate` file and created a test suite for it.

**Comment by otavio** [comment]
@claude Let's open an issue for the edge case for the React UI, but we are not going to add the Vue one because it is being duplicated; instead, we are going to focus on the new UI being developed.
